### PR TITLE
feat(core): stage to depend on an uncorrelated execution

### PR DESCRIPTION
**Pull request by asher**

### CODE REVIEW

**Comment by asher** [comment]
I'm interested in using this with keel (managed delivery) in the context of staggering the regional deployment of multi-region cluster resources. In this scenario, the straight forward path is for keel to emit an execution or executions covering all regions when the initial resource diff has been detected.

This can be handled via a single orchestration chaining deploy and wait stages, but I currently think it would be advantageous to preserve keel's current behavior of one orchestration per resource and region. Launching all at once (but serially, as opposed to doing more sophisticated orchestration within keel) helps with treating the collection of tasks as an atomic unit where actuation remains in progress until all have completed.

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/DependsOnExecutionStage.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+package com.netflix.spinnaker.orca.pipeline;
+
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import com.netflix.spinnaker.orca.pipeline.tasks.DependsOnExecutionTask;
+import javax.annotation.Nonnull;
+
+public class DependsOnExecutionStage implements StageDefinitionBuilder {
+
+  public static String STAGE_TYPE = "dependsOnExecution";
```
Comment (marchello2000) [inline orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/DependsOnExecutionStage.java]: are we doing `STAGE_TYPE` or `PIPELINE_CONFIG_TYPE`?

Either way, probably needs to be `final`

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/tasks/DependsOnExecutionTask.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionNotFoundException;
+import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionRepository;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nonnull;
+import javax.validation.constraints.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DependsOnExecutionTask implements RetryableTask {
```
Comment (marchello2000) [inline orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/tasks/DependsOnExecutionTask.java]: do you want to be able to override timeout in stage config?
```suggestion
public class DependsOnExecutionTask implements OverridableTimeoutRetryableTask {
```

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/DependsOnExecutionStage.java (hunk 1 above)
Comment (asher) [reply orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/DependsOnExecutionStage.java]: @marchello2000 which one should be used? I see both all over.

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/tasks/DependsOnExecutionTask.java (hunk 2 above)
Comment (asher) [reply orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/tasks/DependsOnExecutionTask.java]: Yes, thanks!

**Pull request merged**